### ISSUE TITLE
Content: Ibn Kathir English

### ISSUE THREAD

**Issue by mmahalwy** [quran/quran.com-frontend#309]
http://www.islamwb.com/books/Tafsir%20Ibn%20Kathir%2010%20Volumes.pdf

Does anyone have a parsed version of this or would like to take it on as a project? :)

**Comment by dashohoxha** [comment]
What do you mean by "parsed"? Extracting the text? Converting it to XML/Docbook format? Converting it to JSON? It seems to me like a huge manual work.

**Comment by mmahalwy** [comment]
Probably json would be best, or postgresdb.

Not manual, just need to parse the PDF by searching the titles of the surahs and the individual titles of each section within the surah section.

**Comment by dashohoxha** [comment]
As far as I know, converting PDF to plain text is not 100% reliable (there will always be things that need to be fixed manually). Once this is done, it should not be very difficult to parse the text and split it into sections and subsections. But the first task seems difficult to me (unless you know some way of parsing directly a pdf file).

**Comment by mmahalwy** [comment]
can we do pdf to html? It's easy to go from there

**Comment by dashohoxha** [comment]
It is not clean html: http://dashohoxha.fs.al/tafsir/Tafsir-Ibn-Kathir-10-Volumes.html
The txt is not clean either: http://dashohoxha.fs.al/tafsir/Tafsir-Ibn-Kathir-10-Volumes.txt

**Comment by reshadn** [comment]
Salams. I reached out to alim.org to see if they'd be interested in providing their source/db: http://www.alim.org/library/quran/AlQuran-tafsir/TIK/1/0

**Comment by reshadn** [comment]
Initial reply is that they don't offer source or DB. 
I've asked for a phone meeting to discuss further. 
Should we suggest our own scraping effort? (Is that considered rude?) Should we embed their content directly in our app? I imagine that would increase their server load significantly. 
Any suggestions on the best ways to collaborate with them?

**Comment by mmahalwy** [comment]
@reshadn I think we can suggest scrapping for the sake of turning it into a DB that we can share with them? That might work? Or any help they can give us - for example, html files will make scrapping efforts easier?

**Comment by ATouhou** [comment]
I'll scrape it guys and index it by PAGE number based on the 604 pages mushaf. I will share the database file when I'm done.  
Maybe I can do the "sura + ayah" number correlating page number index table as well, but I think maybe you already got that? (noticed the "line" for each new page in read mode).

**Comment by mmahalwy** [comment]
@ATouhou thanks for that! That'd be amazing. We do have the surah/ayah/line/page/position relationship.

**Comment by reshadn** [comment]
As a follow up the one person I spoke with didn't offer anything. I'm not
sure if that's because they're not clear on who we are or what we're doing.
Or possibly because they don't have bandwidth to follow up.

**Comment by ATouhou** [comment]
@reshadn  I have tried long and hard and explained everything to them detailed. The sourcecode for their webpage is even on github and there is still several exploits left in it, that they have not fixed!
Anyway, I made a scraper and indexed the english tafseer ibn kathir.

I made some split tests and everything looks good :-) 
You can grab the database here:
http://touhou.dk/ibnkathir/tafseer.sql


I might index their other 3 tafaseer,  (all arabic. Ibn kathir, saidi and baghawi)  at some point.
Also i might at somepoint corrolate the ayah chosen (for search) to the right "section" on that page. 
However I am extremely busy with work these days and actually behind deadline on some projects... This was just a side jump, and in Sha Allah one that will be very beneficial for the Ummah! Wal Hamdoullilah. 
Anyways let me know if you have any questions in regards to this or if everything works as I should :-)

You can remove the column "uniqueid and tafsirid", these were for internal purposes if I decide to grab their other tafaseer :p

**Comment by ATouhou** [comment]
Almost forgot,  I can also scrape the resources for word-by-word arabic recitation (and the 2 word connected tajweed rules sounds)  and index it. Data files I already scraped way back :p

**Comment by ATouhou** [comment]
the 3 arabic tafaseer will only some minutes works on the scraper to make it work, give me the headsup if you want / need it :-)

**Comment by mmahalwy** [comment]
@ATouhou where would the audio files come from?

**Comment by ATouhou** [comment]
Wisam Sharif at Recitequran.com
They also have English word-by-word but that is computer generated (if I remember correctly).
Actually we could also use the word-by-word transliteration that the brother Ozair has. (Microsoft guy)

**Comment by mmahalwy** [comment]
@ATouhou i do have the word-by-word arabic from Wisam Sharif but we dont have permissions

**Comment by ATouhou** [comment]
So much for fisabillilah :-) let's change the subject, I'm getting agitated.
Did you check out the SQL file?

**Comment by notacouch** [comment]
@mmahalwy You asked for permission to use Wisam Sharieff and he didn't give? Or couldn't get in touch?

**Comment by ATouhou** [comment]
Permissions should be granted by "Share Islam"
![image](https://cloud.githubusercontent.com/assets/6802820/16878700/d3ee58f0-4aad-11e6-9349-1e7d291bf7e3.png)

 Number (800)409-5044 or email contact@shareislam.com

**Comment by ATouhou** [comment]
I tried through livechat, email (on qtafsir and shareislam),  tried on contactform on recitequran and tried on recitequran.com over a year.  Yielded down to, decision belongs to Yusuf Estetes
![image](https://cloud.githubusercontent.com/assets/6802820/16879038/2b3947b8-4aaf-11e6-9658-d27d7e96383a.png)
![image](https://cloud.githubusercontent.com/assets/6802820/16879095/5ac874cc-4aaf-11e6-8953-6dca6e92f016.png)

**Comment by ATouhou** [comment]
But I did not contact Yusuf Estes on the mail provided. Please feel free to do, and see if you manage to get some permissions :-)

**Comment by notacouch** [comment]
Ah, recitequran.com used to be allahsquran.com/learn, so makes sense it goes back to Sh. Yusuf Estes. 

What are we looking to get from it?

**Comment by ATouhou** [comment]
Just permissions to use their data (audio data), they actually don't need to provide us anything if it will be too much of a hassle for them.  We got the tafseer ibn kathir now, but let us see if Mohammad wants anything else

**Comment by ATouhou** [comment]
@mmahalwy  Which Tafaseer do you have indexed at the moment? I provided the Tafsir Ibn Kathir (English),  do you need anything else? Arabic tafaseer indexed, Or maybe parts of English tafsir as saidi (I think only 1-2 juz are available in english) ?

**Comment by mmahalwy** [comment]
@ATouhou the problem right now is we have tafsir indexed per ayah and ibn kathir is not per ayah. I don't remember which we have indexed but i can check?

**Comment by ATouhou** [comment]
The tafsir ibnu kathir is indexed per page 1-604.  
If you don't already got it (don't think I was given access to the quran.com database) then we could add a table "AyatToPage" where we have a column with page and a column with aya.  then  when looking up tafseer from ibn kathir , we lookup in that table "AyatToPage" for aya say 1:5 , where it will say,  ayah 1:5 is on page 1, and from that we lookup the tafseer. 
![image](https://cloud.githubusercontent.com/assets/6802820/21744139/95ea0d0e-d510-11e6-9c8e-5ea658780be5.png)

**Comment by ATouhou** [comment]
And the HTML includes data on what section to start on at what ayah

**Comment by mmahalwy** [comment]
CC @naveed-ahmad ^^ since he is rebuilding the db from scratch

**Comment by naveed-ahmad** [comment]
@ATouhou can you provide tasfir in json format ? It'll be really easy for me to import json and add it to v3. Following format would be lot of help:

``` json
{ "1": {
   "1": "tasfir",
   "2": "tafisr"
  },
  "2": {
  }
}
```
or

``` json
{ "surah_number1": {
   "ayah1": "tasfir",
   "ayah2": "tafisr"
  },
  "surah2": {
  }
}
```
 
thanks

**Comment by naveed-ahmad** [comment]
Also removes `div` and other html tags, we need to add and index only content no html :)

**Comment by mmahalwy** [comment]
@ATouhou just emailed Sheikh Yusuf. Will update you. CC @ahmedre

**Comment by ATouhou** [comment]
@naveed-ahmad  I just saw your message ~1.5 year late :P Do you still need the data file in another format? I was just following up on some 4000~ tabs in "OneTab" that has been gathering up the last 2 years.

**Comment by AbdullahObaid** [comment]
@naveed-ahmad @mmahalwy This website https://quranenc.com/en/home has many quran Tafsir/Translations which are downloadable in a verse by verse XML format. I hope it can be imported into Quran.com and Quran Android.

**Comment by naveed-ahmad** [comment]
@AbdullahObaid we've already imported their translations, all of em :)

**Comment by nasir-scalestack** [comment]
@naveed-ahmad @AbdullahObaid  just curious how the content is being validated? Not doubting the source but is there a mechanism in place to make sure the source is reliable.

**Comment by ahmedre** [comment]
The quranenc team works with [islamhouse](islamhouse.com), a popular Islamic site, and their translators are transcribing and validating the data. from what i understand, they also have scholars on board to help validate the authenticity of the tafaseer, etc.

**Comment by nasir-scalestack** [comment]
> The quranenc team works with [islamhouse](https://islamhouse.com), a popular Islamic site, and their translators are transcribing and validating the data. from what i understand, they also have scholars on board to help validate the authenticity of the tafaseer, etc.

Awesome.

**Comment by nasir-scalestack** [comment]
Just curious is there a reason my request to contribute to this repo not being acknowledged? I sent an email asking Mohamed and Naveed who my friends here actually know (sheikh Hossam Helal, and Mohamed Al-Hassan) if I can use the database for my own use, and then I realized this was not allowed, so I just got my own translation from tanzil. Regardless I would like to contribute from my 10 years of engineering in Toronto I hope to bring some value, fix some issues, and get the reward from Allah (SWT) in the akhira. Wallahi you guys have a huge blessing with thousands reading it online and I'm extremely jealous I don't have a share of this barakah. You guys will be rewarded even while you are passed from this world because people will benefit from quran.com

I will jump on a skype call if need be, but you have my word and my trust by Allah (SWT) I will only use the resources and intellectualy property strictly for the benefit and use of Quran.com

Jazakallah khair I am eager to help out.

**Comment by ahmedre** [comment]
wa3laikum alsalam,
everyone is free to open PRs and the team tries to be quick about reviewing and merging whatever PRs people open - I checked and don't see any PRs open from you, but I might have missed them. In any case, please also realize we're volunteers - this isn't a full time job for us - so we're unfortunately not always as fast as we'd like to be.

you're more than welcome to open a PR for anything and the team would be more than happy to look at it, give feedback, and accept changes as such.
walsalam 3alaikum.

**Comment by nasir-scalestack** [comment]
Jazakallah khair akhi kareem, I do not have a PR because I assume to actually work and test the frontend I would need an api key to pull the quran.com database from. I definitely understand that everyone is busy and trying their best to contribute.

**Comment by naveed-ahmad** [comment]
waAlakim o salam brother @nasir-scalestack. Just double checked and didn't saw any email from you, when did you sent? 

Anyway, you're always welcome to contribute. There are lots of issues we need to work on, please send your email and ill invite you on slack.

You don't need any key for accessing the api, its free and open for everyone. Here is the api docs: https://quran.api-docs.io/

**Comment by nasir-scalestack** [comment]
Shukran brother Naveed, my apologies may it ended up in the spam folder, my email is nasir@scalestack.io, since no key is required for accessing the api I'll check everything out tonight insha'allah and see where I can contribute. 

Allahuma barak feek.

**Comment by naveed-ahmad** [comment]
Yeah, it was in spam :( invited you to slack channel.

**Comment by masoodhammad** [comment]
I need Quranic Tafsir database  English version. Can anyone help?

**Comment by mimrank** [comment]
any one share **quran tafsir ibn kaseer in json**. we need it in json

**Comment by rizerzero** [comment]
@mimrank Hi, I found this repo, I did not check the data https://github.com/GreentechApps/Al-Quran/tree/master/dbs . 
Otherwise you could scrap websites or APIs, I don't really understand why they refuse to share the database and have one source that everyone could contribute to.
